src/fromager/commands/bootstrap.py

```diff
@@ -72,3 +74,28 @@ def bootstrap(
                 f"removing prebuilt wheel {prebuilt_wheel.name} from download cache"
             )
             filename.unlink()
+
+    constraints_filename = wkctx.work_dir / "constraints.txt"
+    logger.info(f"writing installation dependencies to {constraints_filename}")
+    with open(wkctx.work_dir / "constraints.txt", "w") as f:
+        for name, version in sorted(
+            set(
+                _installation_dependencies(
+                    wkctx.all_edges, context.ROOT_BUILD_REQUIREMENT
+                )
+            )
+        ):
+            f.write(f"{name}=={version}\n")
+
+
+def _installation_dependencies(
+    # The keys of all_edges will be either a validate package name or an empty
+    # string.
+    all_edges: context.BuildRequirements,
+    name: NormalizedName | str,
```

Comment:
Sorry last nit pic since everything is NormalizedName now we can go ahead remove `str`
```suggestion
    name: NormalizedName
```

Reply:
Done